pkg/apis/camel/v1/knative/types.go

```diff
@@ -38,6 +38,7 @@ type CamelServiceDefinition struct {
 	URL         string            `json:"url,omitempty"`
 	Path        string            `json:"path,omitempty"`
 	Metadata    map[string]string `json:"metadata,omitempty"`
+	SinkBinding bool              `json:"sinkBinding,omitempty"`
```

Comment:
I think we need this because the way the logic is defined right now. At a first iteration we select the services. At a second iteration we apply the configuration setting. I think it is cleaner to mark this particular service as a SinkBinding, letting the object to know how to configure later when deserialize, instead of having the controller to assume this logic.

Reply:
My point is that in your code at one point you have a method to mark the service as sink binding, in fact to flip the boolean, then at the end you have a if/then/else but such so in fact the check happens at least 3 times so I wonder if we can avoid this ping pong. 

But this is not a big deal

Reply:
I think the `SetSinkBinding` is only called once, not sure to understand the repeated call you're mentioning. Can you please point in the code?

Reply:
let's do it later, since it is an internal structure